wrappers/wrapper.rb

```diff
@@ -856,19 +863,215 @@ def patch_and_rewind_simplified_constraints(vrp, result)
+          unassigned_exp_ser_count -= result[:unassigned].delete_if{ |uns| uns[:original_service_id] == service.original_id }.size
+          planned_exp_ser_count = simplification_data[:expanded_services].count{ |s| s.original_id == service.original_id } - unassigned_exp_ser_count
+
+          if planned_exp_ser_count == 0
+            # if all of them were unplanned
+            # replace the deleted unassigned expanded ones with one single original un-expanded service
+            result[:unassigned] << {
+              original_service_id: service.original_id,
+              service_id: service.id,
+              point_id: service.activity.point_id,
+              detail: build_detail(service, service.activity, service.activity.point, nil, nil, nil),
+            }
+          else
+            # replace the planned one(s) into one single unexpanded original service
+            first_exp_ser_activity = nil
+            last_exp_ser_activity = nil
+            insert_location = nil
+            deleted_exp_ser_count = 0
+            result[:routes].each{ |route|
+              route[:activities].delete_if.with_index{ |activity, index|
+                if activity[:original_service_id] == service.original_id
+                  insert_location ||= index
+                  first_exp_ser_activity ||= activity # first_exp_ser_activity has the travel and timing info
+                  last_exp_ser_activity = activity # last_exp_ser_activity has the quantity info
+                  deleted_exp_ser_count += 1
+                  true
+                elsif deleted_exp_ser_count == planned_exp_ser_count
+                  break
+                end
+              }
+
+              next unless insert_location # expanded activity(ies) of service is found in this route
+
+              merged_activity = first_exp_ser_activity.merge(last_exp_ser_activity) { |key, first, last|
+                if key == :service_id
+                  service.id
+                elsif key == :detail && first != last
+                  # if only one expanded service is planned (i.e., first == last), then first will be correct
+                  # if not correct the quantities
+                  first[:quantities].each_with_index{ |first_quantity, q_ind|
+                    last_quantity = last[:quantities][q_ind][:unit] == first_quantity[:unit] && last[:quantities][q_ind]
+                    last_quantity ||= last[:quantities].find{ |q| q[:unit] == first_quantity[:unit] }
+                    value_correction = last_quantity[:current_load] - first_quantity[:current_load]
+                    first_quantity[:current_load] = last_quantity[:current_load]
+
+                    next if first_quantity[:value].nil? && value_correction == 0
+
+                    first_quantity[:value] = first_quantity[:value].to_f + value_correction
+                  }
+                  first
+                else
+                  first
+                end
+              }
```

Comment:
Can we trust 100% that all sub-services will be assigned ? I understand that priority is the highest and duration is zero for all sub-services of the set except from the first one. So we will probably assign them all.

However, is it really impossible that any service of this cutting is unassigned ? What if last service of this cutting implies overloading current vehicle capacity ?

Reply:
This is what we decided during daily 
https://gitlab.com/mapotempo/optimizer-api/-/issues/785#note_619768096 

The unplanned sub-services will lead to a warning on the mt-web side:
i. either because only the half of the shipment is done, in that case the planned service will give a relation warning due to unplanned part
ii. or because similar to your example, the optim doesn't do one of the deliveries due to a constraint and mt-web will interpret this as a constraint violataion due to unplanned part (it cannot be capacity because that could be violated during a pickup operation and if we didn't plan the pickup, we cannot plan the delivery so we are the case i. for sure)